query.py

```diff
@@ -88,11 +110,51 @@ def __str__(self):
 class QueryAnd(Query):
     def __init__(self, ops):
         super().__init__()
-        self.ops = ops
-        self.ops_size = {} # Dictionary to hold [op: size]
+        self.ops_size = {}  # Dictionary to hold [op: size]
         self.total_size = None
+        self.ops = ops
+        self.ops = self._flatten_ops()
+
+    def _flatten_ops(self):
+        ops = []
+        for op in self.ops:
+            if isinstance(op, QueryAnd):
+                for _op in op._flatten_ops():
+                    ops.append(_op)
+            else:
+                ops.append(op)
 
-    def evaluate(self, inverted_index):
+        return ops
+
+    def evaluate(self, index, **kwargs):
+        return self._evaluate_set(index)
+        # return self._evaluate(index)
+
+    def _evaluate_set(self, inverted_index):
+        add_ops = [set(op.evaluate(inverted_index)) for op in self.ops if not op.is_flipped]
+        negate_ops = [set(op.evaluate(inverted_index)) for op in self.ops if op.is_flipped]
```

Comment:
Should we sort by size?

Reply:
i tried sorting by size but it had no impact on performance